extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -444,6 +448,7 @@ public Optional<MSQErrorReport> runTask(final Closer closer) throws Exception
   @Override
   public void stopGracefully()
   {
+    log.info("Stopping gracefully");
```

Comment:
Does the logger emit the task ID? If not, it might help to say what is stopping gracefully.